test/offsetarray.jl

```diff
@@ -29,7 +29,10 @@ parenttype{T,N,AA}(::Type{OffsetArray{T,N,AA}}) = AA
 parenttype(A::OffsetArray) = parenttype(typeof(A))
 
 Base.parent(A::OffsetArray) = A.parent
-Base.size(A::OffsetArray) = size(parent(A))
+Base.size(A::OffsetArray) = error("non-certified use of size; audit code and then wrap in @safeindices")
+Base.size(A::OffsetArray, d) = error("non-certified use of size; audit code and then wrap in @safeindices")
+Base.size(s::Base.SafeIndices, A::OffsetArray) = size(s, parent(A))
+Base.size(s::Base.SafeIndices, A::OffsetArray, d) = size(s, parent(A), d)
```

Comment:
I should also clarify that it's up to package authors to exploit this safety, by adding definitions like the 4 above.